src/phoenix/trace/trace_dataset.py

```diff
@@ -199,6 +206,89 @@ def to_disc(self) -> None:
             coerce_timestamps="ms",
         )
 
+    def save(self, directory: Optional[Union[str, Path]] = None) -> UUID:
+        """
+        Writes the trace dataset to disk. If any evaluations have been appended
+        to the dataset, those evaluations will be saved to separate files within
+        the same directory.
+
+        Args:
+            directory (Optional[Union[str, Path]], optional): An optional path
+            to a directory where the data will be written. If not provided, the
+            data will be written to a default location.
+
+        Returns:
+            UUID: The id of the trace dataset, which can be used as key to load
+            the dataset from disk using `load`.
+        """
+        directory = Path(directory or TRACE_DATASET_DIR)
+        for evals in self.evaluations:
+            evals.save(directory)
+        path = directory / TRACE_DATASET_PARQUET_FILE_NAME.format(id=self._id)
+        dataframe = get_serializable_spans_dataframe(self.dataframe)
+        dataframe.to_parquet(
+            path,
+            allow_truncated_timestamps=True,
+            coerce_timestamps="ms",
+        )
+        table = Table.from_pandas(self.dataframe)
+        table = table.replace_schema_metadata(
+            {
+                **(table.schema.metadata or {}),
+                # explicitly encode keys and values, which are automatically encoded regardless
+                b"arize": json.dumps(
+                    {
+                        "dataset_id": str(self._id),
+                        "dataset_name": self.name,
+                        "eval_ids": [str(evals.id) for evals in self.evaluations],
+                    }
+                ).encode("utf-8"),
+            }
+        )
+        parquet.write_table(table, path)
+        return self._id
+
+    @classmethod
+    def load(
+        cls, id: Union[str, UUID], directory: Optional[Union[str, Path]] = None
+    ) -> "TraceDataset":
```

Comment:
we probably need a from_id() constructor function

Reply:
How would that constructor differ from this class method? Currently, `load` can be invoked with `TraceDataset.load(dataset_id)`.